### PR TITLE
✨ Add Dave and Graeme's birthday

### PR DESCRIPTION
**Pull request by EldaveO**
##### Summary

Included Graeme's and Dave's birthday dates
Fixes Issue #643

##### Checklist

- [x] this is a source code change
  - [ ] run `format` in the repository root
  - [ ] run `pytest` in the repository root
  - [x] link to issue being fixed using a [_fixes keyword_](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue), if such an issue exists
  - [ ] update the wiki documentation if necessary
- [ ] or, this is **not** a source code change

### CODE REVIEW

File: duckbot/cogs/announce_day/special_days.py [hunk 1]
```diff
@@ -44,3 +44,5 @@ def _populate(self, year):
         self[date(year, 12, 2)] = f"Female Kelly's Birthday. She's {year-1989} years old"
         self[date(year, 12, 3)] = "DuckBot's Inception Day"
         self[date(year, 12, 5)] = f"Taras' Birthday. He's {year-1989} years old"
+        self[date(year, 12, 15)] = f"Dave's Birthday. He is 29 years old for the {year-2020} in a row"
+        self[date(year, 10, 21)] = f"G-dawg's Birthday. He is a glorious {year-1989} years old!"
```
Comment (jameshughes89) [inline duckbot/cogs/announce_day/special_days.py]: I suggest putting it in order, like the rest of the dates.

**Review by jameshughes89 — CHANGES_REQUESTED** [review]

**Comment by jameshughes89** [comment]
You need to also run `format`. 
image

**Comment by jameshughes89** [comment]
And also run `pytest` for good measure

**Comment by EldaveO** [comment]
> You need to also run `format`. image

How does one those commands?

**Comment by jameshughes89** [comment]
> > You need to also run `format`. image
> 
> How does one those commands?

Do you have it setup locally? Or did you change it in the browser?

If you have it locally, just run the stuff at the top of the README:

```bash
python3.8 -m venv --clear --prompt duckbot venv
. venv/bin/activate
pip install --upgrade pip setuptools wheel
SETUPTOOLS_ENABLE_FEATURES="legacy-editable" pip install --editable .[dev]
```
and then
```bash
pytest  # runs tests, lint and format checks
format  # reformats the entire code base
```

File: duckbot/cogs/announce_day/special_days.py [hunk 2]
```diff
@@ -51,3 +52,4 @@ def _populate(self, year):
         self[date(year, 12, 2)] = f"Female Kelly's Birthday. She's {year-1989} years old"
         self[date(year, 12, 3)] = f"DuckBot's Inception Day. I'm about {(now()-datetime(2020, 12, 3, 10, 39, tzinfo=timezone())).total_seconds()}s old"
         self[date(year, 12, 5)] = f"Taras' Birthday. He's {year-1989} years old"
+        self[date(year, 12, 15)] = f"Dave's Birthday. He is 29 years old for the {year-2020} in a row"
```
Comment (twentylemon) [inline duckbot/cogs/announce_day/special_days.py]: This doesn't really read like english. I like the joke, but the wording needs improvement.

> 2023: _Dave's Birthday. He is 29 years old for the 3 in a row_
> 2024: _Dave's Birthday. He is 29 years old for the 4 in a row_

It should read:

> 2023: _Dave's Birthday. He is 29 years old for the 3rd year in a row_
> 2024: _Dave's Birthday. He is 29 years old for the 4th year in a row_

**Review by twentylemon — CHANGES_REQUESTED** [review]

File: duckbot/cogs/announce_day/special_days.py [hunk 3]
```diff
@@ -43,6 +43,7 @@ def _populate(self, year):
         self[date(year, 9, 26)] = f"Tom's Birthday. He's {year-1990} years old. How he made it this far, we'll never know"
         self[date(year, 10, 7)] = f"Delta's Birthday. He's {year-2015} years old and is a good boy"
         self[date(year, 10, 9)] = "Leif Erikson Day. Hinga Dinga Durgen! https://tenor.com/view/viking-spongebob-squarepants-durgen-fall-down-hard-gif-7302846 "  # intentional trailing space for gif
+        self[date(year, 10, 21)] = f"G-dawg's Birthday. He is a glorious {year-1989} years old!"
```
Comment (twentylemon) [inline duckbot/cogs/announce_day/special_days.py]: Bruh. This obviously isn't g-money's birth year. How could you possibly not be aware of this information.

File: duckbot/cogs/announce_day/special_days.py (hunk 2 above)
Comment (jameshughes89) [reply duckbot/cogs/announce_day/special_days.py]: When discussing offline, @twentylemon suggested creating functionality to make the correct suffix for the number. E.g.

1 -> 1st
2 -> 2nd
3 -> 3rd
everything else 
X -> Xth

**Comment by twentylemon** [comment]
Adding `needs-reply`. This will be automatically closed if there continues to be no traction on it.

File: duckbot/cogs/announce_day/special_days.py (hunk 2 above)
Comment (jameshughes89) [reply duckbot/cogs/announce_day/special_days.py]: Any ideas on where this function would best live? Just in the `SpecialDays` class?

Comment (twentylemon) [reply duckbot/cogs/announce_day/special_days.py]: Left as an exercise to the writer.

**Comment by EldaveO** [comment]
Motivation is coming along. I think I can get some traction on it today.

File: duckbot/cogs/announce_day/special_days.py (hunk 3 above)
Comment (EldaveO) [reply duckbot/cogs/announce_day/special_days.py]: This has been adjusted to 1988.

File: duckbot/cogs/announce_day/special_days.py (hunk 2 above)
Comment (EldaveO) [reply duckbot/cogs/announce_day/special_days.py]: I think to reduce the scope of this pull request I'll make the change to the birthdays and have it working in a reasonable state.

Then I'll make an issue for the adding the ``SpecialDays`` class. This is in an effort to insure we do not miss Graeme's birthday in the future.

Comment (twentylemon) [reply duckbot/cogs/announce_day/special_days.py]: I disagree that the scope of the change need be reduced, it's already quite minimal. Two birthdays and a single function to add a number suffix is not large enough to warrant multiple pull requests.

Comment (jameshughes89) [reply duckbot/cogs/announce_day/special_days.py]: +1

This change would only get merged to _exist_ in an incomplete state before the full change actually occurs.

**Comment by jameshughes89** [comment]
:eyes:

**Comment by Chippers255** [comment]
Bump because I have faith in @EldaveO to finish this.

**Pull request closed without merging**